### PR TITLE
Use Click for CLI Handling

### PR DESCRIPTION
**Pull request by ytausch**
This PR introduces [click](https://github.com/pallets/click) as a new entry point for the CLI. This allows us to make the CLI more consistent, and is a good basis for extending the CLI for local debugging use as described in #2123.

### CODE REVIEW

**Comment by ytausch** [comment]
I am currently adding basic tests that sanity-check the CLI, then this will be ready.

File: conda_forge_tick/deploy.py [hunk 1]
```diff
@@ -161,3 +162,7 @@ def deploy(dry_run=False):
             raise RuntimeError("bot did not push its data! stopping!")
     else:
         print("no files to commit!", flush=True)
+
+
+if __name__ == "__main__":
```
Comment (0xbe7a) [inline conda_forge_tick/deploy.py:167]: Do we want to create a second enty point here? I think having both a central enty point with proper help menues and entry points for each "sub-command" is a bit confusing from a user perspective.

File: conda_forge_tick/deploy.py [hunk 2]
```diff
@@ -14,9 +15,9 @@ def _run_git_cmd(cmd):
     return subprocess.run(cmd, shell=True, check=True)
 
 
-def deploy(dry_run=False):
-    """Deploy the graph to github"""
-    if dry_run:
+def deploy(ctx: CliContext = CliContext()):
```
Comment (0xbe7a) [inline conda_forge_tick/deploy.py:18]: Just commenting on this instance here: Should we provide a default argument here? This makes it easy to "forget" to pass the context and this function is just invoked once with a CliContext anyway

File: conda_forge_tick/lazy_json_backups.py [hunk 3]
```diff
@@ -193,3 +194,7 @@ def _name_to_ts(b):
                 LOGGER.info("saving backup %s", bup)
                 if bup not in curr_fnames:
                     save_backup(bup)
+
+
+if __name__ == "__main__":
+    main_backup()
```
Comment (0xbe7a) [inline conda_forge_tick/lazy_json_backups.py:200]: See entry point comment

File: conda_forge_tick/mappings.py [hunk 4]
```diff
@@ -1,6 +1,7 @@
+from .cli_context import CliContext
 from .pypi_name_mapping import main as main_pypi_name_mapping
 
 
-def main(args):
+def main(_: CliContext = CliContext()):
```
Comment (0xbe7a) [inline conda_forge_tick/mappings.py:5]: I think we can remove this

File: conda_forge_tick/pypi_name_mapping.py [hunk 5]
```diff
@@ -440,5 +441,4 @@ def main(args) -> None:
 
 
 if __name__ == "__main__":
-    # main()
-    pass
+    main()
```
Comment (0xbe7a) [inline conda_forge_tick/pypi_name_mapping.py:444]: See entry point comment

File: conda_forge_tick/update_prs.py [hunk 6]
```diff
@@ -169,17 +171,19 @@ def close_dirty_prs(
 
 
 # @profiling
-def main(args: "CLIArgs") -> None:
-    setup_logger(logger)
+def main(ctx: CliContext = CliContext(), job: int = 1, n_jobs: int = 1) -> None:
```
Comment (0xbe7a) [inline conda_forge_tick/update_prs.py:174]: See entry point

File: conda_forge_tick/status_report.py [hunk 7]
```diff
@@ -501,85 +499,86 @@ def main(args: Any = None) -> None:
...
+    for op in _collect_items_from_nodes(mctx.graph, _get_open_pr_states):
+        open_prs.extend(op)
+    merge_state_count = Counter([o["mergeable_state"] for o in open_prs])
+    with open("./status/pr_state.csv", "a") as f:
+        writer = csv.writer(f)
+        writer.writerow([merge_state_count[k] for k in GH_MERGE_STATE_STATUS])
+    """
 
 
 if __name__ == "__main__":
```
Comment (0xbe7a) [inline conda_forge_tick/status_report.py:584]: See entry point

File: conda_forge_tick/deploy.py (hunk 1 above)
Comment (ytausch) [reply conda_forge_tick/deploy.py]: The main idea behind this was to provide an entry point for local debugging, which is now replaced by the CLI main entry point in bad3bc8f

File: conda_forge_tick/deploy.py (hunk 2 above)
Comment (ytausch) [reply conda_forge_tick/deploy.py]: Makes sense, and also this makes no sense anymore with the new central CLI entry point. Fixed in e97ab2d7

File: conda_forge_tick/lazy_json_backups.py (hunk 3 above)
Comment (ytausch) [reply conda_forge_tick/lazy_json_backups.py]: Fixed in e97ab2d7

File: conda_forge_tick/mappings.py (hunk 4 above)
Comment (ytausch) [reply conda_forge_tick/mappings.py]: Removed in e97ab2d7

File: conda_forge_tick/pypi_name_mapping.py (hunk 5 above)
Comment (ytausch) [reply conda_forge_tick/pypi_name_mapping.py]: Done

File: conda_forge_tick/update_prs.py (hunk 6 above)
Comment (ytausch) [reply conda_forge_tick/update_prs.py]: Removed in e97ab2d7

File: conda_forge_tick/status_report.py (hunk 7 above)
Comment (ytausch) [reply conda_forge_tick/status_report.py]: Done

**Review by beckermr — APPROVED** [review]

**Pull request merged**